lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -87,15 +89,164 @@ class CertHandlerEvents(ObjectEvents):
     cert_changed = EventSource(CertChanged)
 
 
+class _VaultBackend(abc.ABC):
```

Comment:
Should we probably outline the "spec" in a docstring? For example:
```suggestion
class _VaultBackend(abc.ABC):
    """Base class for a single secret manager.
    
    Assumptions:
    - A single secret (label) is managed by a single instance.
    - Secret is per-unit (not per-app, i.e. may differ from unit to unit).
    """
```